circuit/src/gadgets/bits2integer.rs

```diff
@@ -0,0 +1,211 @@
+use crate::{
+	integer::AssignedInteger, params::rns::RnsParams, Chip, Chipset, CommonConfig, FieldExt,
+};
+use halo2::{
+	circuit::{AssignedCell, Layouter},
+	plonk::{Error, Selector},
+};
+
+use super::bits2num::Bits2NumChip;
+
+/// Bits2IntegerChipsetConfig
+#[derive(Debug, Clone)]
+pub struct Bits2IntegerChipsetConfig {
+	bits2num: Selector,
+}
+
+impl Bits2IntegerChipsetConfig {
+	/// Construct a new config.
+	pub fn new(bits2num: Selector) -> Self {
+		Self { bits2num }
+	}
+}
+
+/// Constructs a cell and a variable for the circuit.
+#[derive(Clone)]
+pub struct Bits2IntegerChipset<
+	W: FieldExt,
+	N: FieldExt,
+	const NUM_LIMBS: usize,
+	const NUM_BITS: usize,
+	P,
+> where
+	P: RnsParams<W, N, NUM_LIMBS, NUM_BITS>,
+{
+	/// Assigned Integer variable.
+	assigned_integer: AssignedInteger<W, N, NUM_LIMBS, NUM_BITS, P>,
+}
+
+impl<W: FieldExt, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P>
+	Bits2IntegerChipset<W, N, NUM_LIMBS, NUM_BITS, P>
+where
+	P: RnsParams<W, N, NUM_LIMBS, NUM_BITS>,
+{
+	/// Create a new chip.
+	pub fn new(assigned_integer: AssignedInteger<W, N, NUM_LIMBS, NUM_BITS, P>) -> Self {
+		Self { assigned_integer }
+	}
+}
+
+impl<W: FieldExt, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P> Chipset<N>
+	for Bits2IntegerChipset<W, N, NUM_LIMBS, NUM_BITS, P>
+where
+	P: RnsParams<W, N, NUM_LIMBS, NUM_BITS>,
+{
+	type Config = Bits2IntegerChipsetConfig;
+	type Output = [Vec<AssignedCell<N, N>>; NUM_LIMBS];
```

Comment:
We should merge all the bits here and return one big array

Reply:
If we go with limbs, each limb should be 68 bits, and we check them individually, so it will never wrap around the field

Reply:
> If we want to return one vector which carries the composed field value of the limbs
We are not returning composed value, instead we are returning bits that we get from Bits2Num